staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1365,9 +1367,36 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		)
 	}
 
+	if opts.CountMetricPollPeriod > 0 {
+		stopFunc := e.startObservingCount(opts.CountMetricPollPeriod)
+		previousDestroy := e.DestroyFunc
+		e.DestroyFunc = func() {
+			stopFunc()
+			if previousDestroy != nil {
+				previousDestroy()
+			}
+		}
+	}
+
 	return nil
 }
 
+// startObservingCount starts monitoring given prefix and periodically updating metrics. It return a function to stop collection.
+func (e *Store) startObservingCount(period time.Duration) func() {
+	prefix := e.KeyRootFunc(genericapirequest.NewContext())
+	resourceName := e.DefaultQualifiedResource.String()
+	glog.V(2).Infof("Monitoring %v bound to <storage-prefix>/%v", resourceName, prefix)
+	stopCh := make(chan struct{})
+	go wait.Until(func() {
+		count, err := e.Storage.Count(prefix)
+		if err != nil {
+			glog.V(10).Infof("Failed to update storage count metric: %v", err)
+		}
+		metrics.UpdateObjectCount(resourceName, count)
```

Comment:
do you still want to update the count if there was an error fetching?

Reply:
Yes, keeping old value might be misleading. 0 should be informative enough, but I can change it to -1 if you think it'll be better.

Reply:
distinguishing between an error condition and "all my objects were deleted" in the metric is probably valuable :)

Reply:
This is already a polled value, so there's some expectation that it's approximate. Does prometheus have any notion of "last updated"?

Reply:
I don't believe it does. It just exposes a value which is then scraped by the backend. As long as the value is possible to scrape it'll treat it as is. It does have a logic for dealing with missing data, but I don't know of any way of saying "mark this data as missing".

Reply:
> It does have a logic for dealing with missing data, but I don't know of any way of saying "mark this data as missing".

Maybe removing the value ([DeleteLabelValues](https://godoc.org/github.com/prometheus/client_golang/prometheus#CounterVec.DeleteLabelValues))?